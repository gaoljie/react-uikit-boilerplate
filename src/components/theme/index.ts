import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
