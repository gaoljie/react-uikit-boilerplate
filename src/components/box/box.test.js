import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";
import theme from "../theme";
import Box from "./index";

const props = {
  as: "p",
  dp: "flex",
  fd: "column",
  bs: "border-box",
  width: "180px",
  height: "100px",
  fs: "18px",
  fw: "wrap",
  jc: "flex-start",
  ai: "flex-end",
  ji: "start",
  bg: "white",
  m: "10px",
  ta: "center",
  lh: "1",
  color: "black",
  p: "10px",
  ml: "20px",
  mr: "21px",
  mt: "22px",
  mb: "23px",
  pl: "24px",
  pr: "25px",
  pt: "26px",
  pb: "27px",
  children: <span>children</span>
};
describe("Test Box", () => {
  it("Matches the snapshot", () => {
    const wrapper = mount(<Box {...props}>content</Box>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Styling", () => {
    const wrapper = mount(<Box {...props} />);

    expect(wrapper.exists(props.as)).toEqual(true);

    expect(wrapper.props().children).toEqual(props.children);

    expect(wrapper).toHaveStyleRule("display", props.dp);
    expect(wrapper).toHaveStyleRule("color", props.color);
    expect(wrapper).toHaveStyleRule("font-size", props.fs);
    expect(wrapper).toHaveStyleRule("text-align", props.ta);
    expect(wrapper).toHaveStyleRule("flex-direction", props.fd);
    expect(wrapper).toHaveStyleRule("line-height", props.lh);
    expect(wrapper).toHaveStyleRule("box-sizing", props.bs);
    expect(wrapper).toHaveStyleRule("width", props.width);
    expect(wrapper).toHaveStyleRule("height", props.height);
    expect(wrapper).toHaveStyleRule("justify-content", props.jc);
    expect(wrapper).toHaveStyleRule("align-items", props.ai);
    expect(wrapper).toHaveStyleRule("justify-items", props.ji);
    expect(wrapper).toHaveStyleRule("background", props.bg);
    expect(wrapper).toHaveStyleRule("flex-wrap", props.fw);
    expect(wrapper).toHaveStyleRule("margin", props.m);
    expect(wrapper).toHaveStyleRule("padding", props.p);
    expect(wrapper).toHaveStyleRule("margin-left", props.ml);
    expect(wrapper).toHaveStyleRule("margin-right", props.mr);
    expect(wrapper).toHaveStyleRule("margin-top", props.mt);
    expect(wrapper).toHaveStyleRule("margin-bottom", props.mb);
    expect(wrapper).toHaveStyleRule("padding-left", props.pl);
    expect(wrapper).toHaveStyleRule("padding-right", props.pr);
    expect(wrapper).toHaveStyleRule("padding-top", props.pt);
    expect(wrapper).toHaveStyleRule("padding-bottom", props.pb);

    wrapper.setProps({
      height: {
        xl: "100px"
      }
    });

    expect(wrapper).toHaveStyleRule("height", `100px`);

    wrapper.setProps({
      height: {
        lg: "120px"
      }
    });

    expect(wrapper).toHaveStyleRule("height", `120px`, {
      media: theme.smallDesktop.slice(7)
    });

    wrapper.setProps({
      height: {
        md: "140px"
      }
    });

    expect(wrapper).toHaveStyleRule("height", `140px`, {
      media: theme.tablet.slice(7)
    });

    wrapper.setProps({
      height: {
        sm: "160px"
      }
    });

    expect(wrapper).toHaveStyleRule("height", `160px`, {
      media: theme.mobile.slice(7)
    });

    wrapper.setProps({ center: true });
    expect(wrapper).toHaveStyleRule("display", `grid`);
    expect(wrapper).toHaveStyleRule("place-items", `center`);

    wrapper.setProps({ vCenter: true });
    expect(wrapper).toHaveStyleRule("align-items", `center`);

    wrapper.setProps({ hCenter: true });
    expect(wrapper).toHaveStyleRule("justify-items", `center`);

    expect(wrapper).not.toHaveStyleRule("grid-column", `span 1`);

    wrapper.setProps({ xl: 4, lg: 3, md: 2, sm: 2 });

    expect(wrapper).toHaveStyleRule("grid-column", `span 4`);

    expect(wrapper).toHaveStyleRule("grid-column", `span 3`, {
      media: theme.smallDesktop.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `span 2`, {
      media: theme.tablet.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `span 2`, {
      media: theme.mobile.slice(7)
    });

    wrapper.setProps({ start: 2 });

    expect(wrapper).toHaveStyleRule("grid-column", `2 / span 4`);

    expect(wrapper).toHaveStyleRule("grid-column", `2 / span 3`, {
      media: theme.smallDesktop.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `2 / span 2`, {
      media: theme.tablet.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `2 / span 2`, {
      media: theme.mobile.slice(7)
    });

    wrapper.setProps({
      start: {
        xl: 3
      }
    });

    expect(wrapper).toHaveStyleRule("grid-column", `3 / span 4`);

    expect(wrapper).toHaveStyleRule("grid-column", `3 / span 3`, {
      media: theme.smallDesktop.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `3 / span 2`, {
      media: theme.tablet.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `3 / span 2`, {
      media: theme.mobile.slice(7)
    });

    wrapper.setProps({
      start: {
        lg: 4
      }
    });

    expect(wrapper).toHaveStyleRule("grid-column", `4 / span 3`, {
      media: theme.smallDesktop.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `4 / span 2`, {
      media: theme.tablet.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `4 / span 2`, {
      media: theme.mobile.slice(7)
    });

    wrapper.setProps({
      start: {
        md: 5
      }
    });

    expect(wrapper).toHaveStyleRule("grid-column", `5 / span 2`, {
      media: theme.tablet.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-column", `5 / span 2`, {
      media: theme.mobile.slice(7)
    });

    wrapper.setProps({
      start: {
        sm: 6
      }
    });

    expect(wrapper).toHaveStyleRule("grid-column", `6 / span 2`, {
      media: theme.mobile.slice(7)
    });
  });
});
