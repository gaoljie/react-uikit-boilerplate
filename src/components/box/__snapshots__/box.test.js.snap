// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Box Matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 18px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  width: 180px;
  height: 100px;
  color: black;
  line-height: 1;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  justify-items: start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background: white;
  margin: 10px;
  padding: 10px;
  text-align: center;
  margin-top: 22px;
  margin-left: 20px;
  margin-right: 21px;
  margin-bottom: 23px;
  padding-top: 26px;
  padding-left: 24px;
  padding-right: 25px;
  padding-bottom: 27px;
}

<styled.div
  ai="flex-end"
  as="p"
  bg="white"
  bs="border-box"
  color="black"
  dp="flex"
  fd="column"
  fs="18px"
  fw="wrap"
  height="100px"
  jc="flex-start"
  ji="start"
  lh="1"
  m="10px"
  mb="23px"
  ml="20px"
  mr="21px"
  mt="22px"
  p="10px"
  pb="27px"
  pl="24px"
  pr="25px"
  pt="26px"
  ta="center"
  width="180px"
>
  <StyledComponent
    ai="flex-end"
    as="p"
    bg="white"
    bs="border-box"
    color="black"
    dp="flex"
    fd="column"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    fs="18px"
    fw="wrap"
    height="100px"
    jc="flex-start"
    ji="start"
    lh="1"
    m="10px"
    mb="23px"
    ml="20px"
    mr="21px"
    mt="22px"
    p="10px"
    pb="27px"
    pl="24px"
    pr="25px"
    pt="26px"
    ta="center"
    width="180px"
  >
    <p
      className="c0"
      color="black"
      height="100px"
      width="180px"
    >
      content
    </p>
  </StyledComponent>
</styled.div>
`;
